xray/test/test_dataarray.py

```diff
@@ -382,6 +382,64 @@ def test_sel_method(self):
         actual = data.sel(x=[0.9, 1.9], method='backfill')
         self.assertDataArrayIdentical(expected, actual)
 
+    def test_isel_points_method(self):
+        shape = (10, 5, 6)
+        np_array = np.random.random(shape)
+        da = DataArray(np_array, dims=['time', 'y', 'x'])
+        y = [1, 3]
+        x = [3, 0]
+
+        expected = da.values[:, y, x]
+
+        actual = da.isel_points(y=y, x=x, dim='test_coord')
+        assert 'test_coord' in actual.coords
+        assert actual.coords['test_coord'].shape == (len(y), )
+
+        actual = da.isel_points(y=y, x=x)
+        assert 'points' in actual.coords
+        # not sure why actual needs to be transposed
```

Comment:
xray always concatenates along the first dimension.

NumPy tries to do something very clever with fancy indexing and concatenates along the last dimension in this case:
```
In [11]: x[:, [1, 3], [3, 0]].shape
Out[11]: (10, 2)
```